### PR TITLE
doc: add overlay network to overview section

### PR DESCRIPTION
**Pull request by garlick**
Problem: we don't have any documentation to help with the selection of overlay network topology.

Add a section to the admin guide "Overview and Background" section.

### CODE REVIEW

File: doc/guide/admin.rst [hunk 1]
```diff
@@ -249,6 +249,70 @@ See also:
...
+
+Topology for Large Clusters
+---------------------------
+
+To avoid the above downsides, larger clusters should use a custom topology
+with tree height of 2 and internal brokers assigned to stable, well connected,
+non-busy nodes.  The downside of this topology is, obviously:
+
+- More brokers are critical
+
```
Comment (grondo) [inline doc/guide/admin.rst:315]: Suggestion: add a short descriptoin about how to use the `bootstrap` config to create the two-level hierarchy and reference `flux-config-bootstrap(5)`? (Or maybe just reference the 3rd example in that manual)

**Review by grondo — APPROVED** [review]
Great! This is helpful! Just one take-it-or-leave-it suggestion inline.

File: doc/guide/admin.rst (hunk 1 above)
Comment (grondo) [reply doc/guide/admin.rst:315]: I didn't reread the whole document, so perhaps that would be redundant. Ignore me if so

Comment (garlick) [reply doc/guide/admin.rst:315]: Sounds like a good idea.  That will probably make all the words more directly useful.

**Comment by garlick** [comment]
Thanks!  I added those references to the bootstrap examples and will set MWP.

**Pull request merged**